spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/MemoryData.scala

```diff
@@ -128,14 +152,16 @@ case class HeapData(private var _shape: Array[Int], private var _layout: Int) ex
     s"HeapData([${shape.mkString("x")}], ${layout})"
   }
 
-  override def cloneFormat(): MemoryData = new HeapData(_shape, _layout)
+  override def cloneFormat(): MemoryData = new HeapData(_shape, _layout, _dataType)
```

Comment:
func name is not clear.

Reply:
Yes. but it's a legacy code. I will put the refactor into another pr.